xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiHibernateVersioningStore.java

```diff
@@ -111,18 +108,23 @@ public XWikiHibernateVersioningStore()
 
     @Override
     public Version[] getXWikiDocVersions(XWikiDocument doc, XWikiContext context) throws XWikiException
+    {
+        return getXWikiDocVersions(doc, new RevisionCriteria(null, null, null, true), context).toArray(new Version[0]);
+    }
+
+    @Override
+    public Collection<Version> getXWikiDocVersions(XWikiDocument doc, RevisionCriteria criteria, XWikiContext context)
+        throws XWikiException
     {
         try {
-            XWikiDocumentArchive archive = getXWikiDocumentArchive(doc, context);
+            XWikiDocumentArchive archive = getXWikiDocumentArchive(doc, criteria, context);
             if (archive == null) {
-                return new Version[0];
+                return Collections.emptyList();
```

Comment:
Slight shorted version based on Java standard APIs.
```suggestion
                return List.of();
```